engine/src/main/java/org/terasology/engine/subsystem/lwjgl/Lwjgl2Sync.java

```diff
@@ -0,0 +1,163 @@
+     */
+    public static void sync(int fps) {
+        if (fps <= 0) return;
+        if (!initialised) initialise();
+
+        try {
+            // sleep until the average sleep time is greater than the time remaining till nextFrame
+            for (long t0 = getTime(), t1; (nextFrame - t0) > sleepDurations.avg(); t0 = t1) {
+                Thread.sleep(1);
+                sleepDurations.add((t1 = getTime()) - t0); // update average sleep time
+            }
+
+            // slowly dampen sleep average if too high to avoid yielding too much
+            sleepDurations.dampenForLowResTicker();
+
+            // yield until the average yield time is greater than the time remaining till nextFrame
+            for (long t0 = getTime(), t1; (nextFrame - t0) > yieldDurations.avg(); t0 = t1) {
+                Thread.yield();
+                yieldDurations.add((t1 = getTime()) - t0); // update average yield time
+            }
+        } catch (InterruptedException e) {
+
+        }
+
+        // schedule next frame, drop frame(s) if already too late for next frame
+        nextFrame = Math.max(nextFrame + NANOS_IN_SECOND / fps, getTime());
+    }
+
+    /**
+     * This method will initialise the sync method by setting initial values for sleepDurations/yieldDurations and
+     * nextFrame.
+     * <p>
+     * If running on windows it will start the sleep timer fix.
+     */
+    private static void initialise() {
+        initialised = true;
+
+        sleepDurations.init(1000 * 1000);
+        yieldDurations.init((int) (-(getTime() - getTime()) * 1.333));
+
+        nextFrame = getTime();
+
+        String osName = System.getProperty("os.name");
+
+        if (osName.startsWith("Win")) {
+            // On windows the sleep functions can be highly inaccurate by
+            // over 10ms making in unusable. However it can be forced to
+            // be a bit more accurate by running a separate sleeping daemon
+            // thread.
+            Thread timerAccuracyThread = new Thread(new Runnable() {
+                public void run() {
+                    try {
+                        Thread.sleep(Long.MAX_VALUE);
+                    } catch (Exception e) {
+                    }
+                }
+            });
+
+            timerAccuracyThread.setName("LWJGL Timer");
+            timerAccuracyThread.setDaemon(true);
+            timerAccuracyThread.start();
+        }
+    }
+
+    /**
+     * Get the system time in nano seconds
+     *
+     * @return will return the current time in nano's
+     */
+    private static long getTime() {
+        return (long) (GLFW.glfwGetTime() * NANOS_IN_SECOND);
+    }
+
+    private static class RunningAvg {
+        private final long[] slots;
+        private int offset;
+
+        private static final long DAMPEN_THRESHOLD = 10 * 1000L * 1000L; // 10ms
+        private static final float DAMPEN_FACTOR = 0.9f; // don't change: 0.9f is exactly right!
```

Comment:
Howcome! 😁

Reply:
Copy-paste from LWJGL2's class.
It is Sync.java - it is exactly right! :D